rest/route/host_routes.go

```diff
@@ -328,3 +335,271 @@ func (hph *hostPostHandler) Execute(ctx context.Context, sc data.Connector) (Res
+}
+
+func getHostChangeRDPPasswordRouteManager(route string, version int) *RouteManager {
+	return &RouteManager{
+		Route:   route,
+		Version: version,
+		Methods: []MethodHandler{
+			{
+				PrefetchFunctions: []PrefetchFunc{PrefetchUser},
+				MethodType:        http.MethodPost,
+				Authenticator:     &RequireUserAuthenticator{},
+				RequestHandler:    &hostChangeRDPPasswordHandler{},
+			},
+		},
+	}
+}
+
+type hostChangeRDPPasswordHandler struct {
+	hostID      string
+	rdpPassword string
+}
+
+func (h *hostChangeRDPPasswordHandler) Handler() RequestHandler {
+	return &hostChangeRDPPasswordHandler{}
+}
+
+func (h *hostChangeRDPPasswordHandler) ParseAndValidate(ctx context.Context, r *http.Request) error {
+	hostModify := model.APISpawnHostModify{}
+	if err := util.ReadJSONInto(util.NewRequestReader(r), &hostModify); err != nil {
+		return err
+	}
+
+	var err error
+	h.hostID, err = validateHostID(mux.Vars(r)["host_id"])
+	if err != nil {
+		return err
+	}
+
+	h.rdpPassword = string(hostModify.RDPPwd)
+	if !spawn.ValidateRDPPassword(h.rdpPassword) {
+		return &rest.APIError{
+			StatusCode: http.StatusBadRequest,
+			Message:    "invalid password",
+		}
+	}
+
+	return nil
+}
+
+func (h *hostChangeRDPPasswordHandler) Execute(ctx context.Context, sc data.Connector) (ResponseData, error) {
+	u := MustHaveUser(ctx)
+
+	host, err := fetchHostAndAuthorize(sc, h.hostID, u)
+	if err != nil {
+		return ResponseData{}, err
+	}
+
+	if !host.Distro.IsWindows() {
+		return ResponseData{}, &rest.APIError{
+			StatusCode: http.StatusBadRequest,
+			Message:    "RDP passwords can only be set on Windows hosts",
+		}
+	}
+	if host.Status != evergreen.HostRunning {
+		return ResponseData{}, &rest.APIError{
+			StatusCode: http.StatusBadRequest,
+			Message:    "RDP passwords can only be set on running hosts",
+		}
+	}
+
+	if err := sc.SetHostPassword(ctx, host, h.rdpPassword); err != nil {
+		return ResponseData{}, &rest.APIError{
+			StatusCode: http.StatusInternalServerError,
+			Message:    err.Error(),
+		}
+	}
+
+	return ResponseData{}, nil
+}
+
+func getHostExtendExpirationRouteManager(route string, version int) *RouteManager {
+	return &RouteManager{
+		Route:   route,
+		Version: version,
+		Methods: []MethodHandler{
+			{
+				PrefetchFunctions: []PrefetchFunc{PrefetchUser},
+				MethodType:        http.MethodPost,
+				Authenticator:     &RequireUserAuthenticator{},
+				RequestHandler:    &hostExtendExpirationHandler{},
+			},
+		},
+	}
+}
+
+type hostExtendExpirationHandler struct {
+	hostID   string
+	addHours int
+}
+
+func (h *hostExtendExpirationHandler) Handler() RequestHandler {
+	return &hostExtendExpirationHandler{}
+}
+
+func (h *hostExtendExpirationHandler) ParseAndValidate(ctx context.Context, r *http.Request) error {
+	hostModify := model.APISpawnHostModify{}
+	if err := util.ReadJSONInto(util.NewRequestReader(r), &hostModify); err != nil {
+		return err
+	}
+
+	var err error
+	h.hostID, err = validateHostID(mux.Vars(r)["host_id"])
```

Comment:
We should also validate that the host is not terminated

Reply:
done